relay/kusama/src/lib.rs

```diff
@@ -1807,6 +1890,66 @@ impl pallet_asset_rate::Config for Runtime {
 	type BenchmarkHelper = polkadot_runtime_common::impls::benchmarks::AssetRateArguments;
 }
 
+// Derived from `kusama_asset_hub_runtime::RuntimeBlockWeights`.
+const AH_MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(
+	frame_support::weights::constants::WEIGHT_REF_TIME_PER_SECOND.saturating_mul(2),
+	polkadot_primitives::MAX_POV_SIZE as u64,
+);
+
+parameter_types! {
+	// Exvivalent to `kusama_asset_hub_runtime::MessageQueueServiceWeight`.
+	pub AhMqServiceWeight: Weight = Perbill::from_percent(50) * AH_MAXIMUM_BLOCK_WEIGHT;
+	// 80 percent of the `AhMqServiceWeight` to leave some space for XCM message base processing.
+	pub AhMigratorMaxWeight: Weight = Perbill::from_percent(80) * AhMqServiceWeight::get();
+	pub RcMigratorMaxWeight: Weight = Perbill::from_percent(60) * BlockWeights::get().max_block;
+	pub AhExistentialDeposit: Balance = EXISTENTIAL_DEPOSIT / 100;
+	pub const XcmResponseTimeout: BlockNumber = 30 * DAYS;
+	pub const AhUmpQueuePriorityPattern: (BlockNumber, BlockNumber) = (18, 2);
+}
+
+pub struct ProxyTypeAny;
+impl frame_support::traits::Contains<TransparentProxyType> for ProxyTypeAny {
+	fn contains(proxy_type: &TransparentProxyType) -> bool {
+		proxy_type.0 == kusama_runtime_constants::proxy::ProxyType::Any
+	}
+}
+
+impl pallet_rc_migrator::Config for Runtime {
+	type RuntimeOrigin = RuntimeOrigin;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeHoldReason = RuntimeHoldReason;
+	type RuntimeFreezeReason = RuntimeFreezeReason;
+	type RuntimeEvent = RuntimeEvent;
+	type AdminOrigin = EitherOfDiverse<
+		EnsureRoot<AccountId>,
+		EitherOfDiverse<Fellows, EnsureXcm<Equals<AssetHubLocation>, Location>>,
+	>;
+	type Currency = Balances;
+	type CheckingAccount = xcm_config::CheckAccount;
+	type TreasuryBlockNumberProvider = System;
+	type TreasuryPaymaster = TreasuryPaymaster;
+	type PureProxyFreeVariants = ProxyTypeAny;
+	type SendXcm = xcm_config::XcmRouterWithoutException;
+	type MaxRcWeight = RcMigratorMaxWeight;
+	type MaxAhWeight = AhMigratorMaxWeight;
+	type AhExistentialDeposit = AhExistentialDeposit;
+	type RcWeightInfo = weights::pallet_rc_migrator::WeightInfo<Runtime>;
+	type AhWeightInfo = weights::pallet_ah_migrator::WeightInfo<ah_migration::weights::AhDbConfig>;
+	type RcIntraMigrationCalls = ahm_phase1::CallsEnabledDuringMigration;
+	type RcPostMigrationCalls = ahm_phase1::CallsEnabledAfterMigration;
+	type StakingDelegationReason = ahm_phase1::StakingDelegationReason;
```

Comment:
This reads to me that delegated staking was the only pallet that was being migrated that used Hold and therefore needs a reason. But, now `staking` (on RC) also uses holds. Is this handled as well? 

I guess it is handled, but makes me sus because they are being handled in _different_ ways. I would have expected to see a hold reason for staking here as well.

Reply:
cc @muharem looks like we use it in `get_provider_count`. Can you elaborate please and maybe put a comment on the type?

FWIW I see this in the md:  
```pre
- delegate_staking (P/K/W): One extra provider reference should be migrated to AH for every account
with the hold reason `pallet_delegated_staking::HoldReason::StakingDelegation`. This ensures the
entire balance, including the ED, can be staked via holds.
Source: <https://github.com/paritytech/polkadot-sdk/blob/ab1e12ab6f6c3946c3c61b97328702e719cd1223/substrate/frame/delegated-staking/src/types.rs#L81>

Reply:
yeah you are right, my assumption was wrong -- staking holds are managed correctly. This special reason is for the provider fixing. I will look more closely into how all holds + providers are managed in migration, but so far looks good.